packages/react-reconciler/src/ReactFiberHydrationContext.js

```diff
@@ -109,11 +113,22 @@ function insertNonHydratedInstance(returnFiber: Fiber, fiber: Fiber) {
           case HostComponent:
             const type = fiber.type;
             const props = fiber.pendingProps;
-            didNotFindHydratableContainerInstance(parentContainer, type, props);
+            didNotFindHydratableContainerInstance(
+              parentContainer,
+              type,
+              props,
+              fiber.index,
```

Comment:
I briefly talked to @sebmarkbage and he pointed out this doesn't necessarily give you the index inside the parent's host nodes.

Specifically, `fiber.index` is a position in *any* parent Fiber, which could be a fragment (or even a nested fragment), it could come after something that renders `null` (and thus doesn't produce a node), etc.

So this won't work. I'm not sure what the solution would be, but at least this gives you a starting point with missing test cases. 🙃

Reply:
All right, thank you, this is [exactly what I asked about when I just started](https://github.com/facebook/react/pull/12063#discussion_r171792028) because this assumption about `fiber.index` was the foundation for the diff display.

Reply:
Sorry 🤦‍♂️

Reply:
Here's the failing test I could come up with where the indexes are off. https://github.com/facebook/react/pull/12063/commits/7d6bbdef4086736c0461ce734b0bba460d96f0eb